src/dataflow/components/dataflow-program.tsx

```diff
@@ -284,14 +295,19 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
       this.programEditor.view.resize();
       this.programEditor.trigger("process");
 
-      this.intervalHandle = setInterval(this.heartBeat, HEARTBEAT_INTERVAL);
-
-      if (this.getRunState() !== ProgramRunStates.Ready) {
-        this.setState({programRunState: this.getRunState(), showGraph: true});
+      const programRunState: ProgramRunStates = this.getRunState();
+      if (programRunState !== ProgramRunStates.Ready) {
+        const hasDataStorage = this.getNodeCount("Data Storage") > 0;
+        const programDisplayState = hasDataStorage ? ProgramDisplayStates.Graph : ProgramDisplayStates.Program;
```

Comment:
It's my understanding that this is the desired behavior.  It might change when the DF team sees it in action, but they want the full tile graph mode to have precedence.